hibernate-community-dialects/src/main/java/org/hibernate/community/dialect/InformixDialect.java

```diff
@@ -466,7 +466,7 @@ protected <T extends JdbcOperation> SqlAstTranslator<T> buildTranslator(
 	@Override
 	public String extractPattern(TemporalUnit unit) {
 		return switch ( unit ) {
-			case SECOND -> "to_number(to_char(?2,'%S.%F3'))";
+			case SECOND -> getVersion().isBefore( 11, 70 ) ?	"to_number(to_char(?2,'%S%F3'))" : "to_number(to_char(?2,'%S.%F3'))";
```

Comment:
Would you mind adding a javadoc comment that explains why this is necessary? Do older Informix versions not support the `.` character? I know some databases need special quoting for non-pattern characters with e.g. double quote i.e. `"."`. Maybe doing the quoting makes this work? Or are older versions appending the `.` char automatically for `%F3`?